app/src/main/java/com/yml/womensafety/navigationdrawer/contacts/ContactsFragment.kt

```diff
@@ -98,9 +85,26 @@ class ContactsFragment : Fragment(R.layout.fragment_contacts) {
             )
         }
         if (contactId != null) {
-            ref.child(contactId).setValue(contact).addOnCompleteListener {
-                Toast.makeText(view?.context, "Success", Toast.LENGTH_LONG).show()
-            }
+            databaseReference.orderByChild("contactNumber").equalTo(contactNumber.text.toString())
+                .addListenerForSingleValueEvent(object : ValueEventListener {
+                    override fun onDataChange(snapshot: DataSnapshot) {
+                        if (snapshot.exists()) {
+                            contactNumber.error = "Number already exists"
+                            contactNumber.requestFocus()
+                        }
+                        else {
+                            databaseReference.child(contactId).setValue(contact).addOnCompleteListener {
+                                Toast.makeText(view?.context, "Success", Toast.LENGTH_LONG).show()
+                            }
+                        }
+                    }
+
+                    override fun onCancelled(error: DatabaseError) {
+
```

Comment:
Add comments in the empty methods as to why they are empty

Reply:
Added Toast message to display the error

Reply:
Try using an alert dialog to show error. That is a better user experience